core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,929 @@ def update_question_suggestion(
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionReviewStatsModel.put_multi(
+        stats_models_to_update)
+
+
+def get_contribution_date(date: datetime.datetime) -> datetime.date:
+    """Gets the string value of the given date.
+
+    Args:
+        date: datetime.datetime. Date that needs to converted into a string.
+
+    Returns:
+        datetime.date. The formatted object of the given date.
+    """
+    return datetime.datetime.strptime(str(
+                date.utcnow().date().isoformat()),
+                '%Y-%m-%d'
+            ).date()
+
+
+def update_translation_contribution_stats_at_submission(
+    suggestion: suggestion_registry.BaseSuggestion,
+    user_id: str, target_id: str
+) -> None:
+    """Creates/updates TranslationContributionStatsModel model for
+    given translation submitter when a translation is submitted.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            submitted.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+    """
+    topic_id = ''
+    exp_opportunity = (
+        opportunity_services.get_exploration_opportunity_summary_by_id(
+            target_id))
+    # We can confirm that exp_opportunity will not be None since there should
+    # be an assigned opportunity for a given translation. Hence we can rule out
+    # the possibility of None for mypy type checking.
+    assert exp_opportunity is not None
+    topic_id = exp_opportunity.topic_id
+
+    content_plain_text = html_cleaner.strip_html_tags(
+        suggestion.change.translation_html)
+    content_word_count = len(content_plain_text.split())
+
+    translation_contribution_stat_model = (
+        suggestion_models.TranslationContributionStatsModel.get(
+            suggestion.change.language_code, user_id, topic_id
+        ))
+
+    if translation_contribution_stat_model is None:
+        suggestion_models.TranslationContributionStatsModel.create(
+            language_code=suggestion.change.language_code,
+            contributor_user_id=user_id,
+            topic_id=topic_id,
+            submitted_translations_count=1,
+            submitted_translation_word_count=content_word_count,
+            accepted_translations_count=0,
+            accepted_translations_without_reviewer_edits_count=0,
+            accepted_translation_word_count=0,
+            rejected_translations_count=0,
+            rejected_translation_word_count=0,
+            contribution_dates=[get_contribution_date(suggestion.last_updated)]
+        )
+    else:
+        translation_contribution_stat = (
+            create_translation_contribution_stats_from_model(
+                translation_contribution_stat_model))
+
+        get_incremental_translation_contribution_stats(
+            translation_contribution_stat, content_word_count,
+            suggestion.last_updated)
```

Comment:
This particular function seems simple enough to just inline here.

Reply:
Done